### DISCUSSION TITLE
How to verify compilation status when compiling through custom command?

### DISCUSSION THREAD

**Discussion by rishabkaushik** [codefori/vscode-ibmi#1821] [Q&A] [answered]
We have in-house source management system which has it's own custom commands for compilation. Our command submits a job to create a module first and then create a program object. Since the command submits the job and does not return a status code if any compilation error was there, there are no errors shown inline in the source view. Below is the sample command.

`COMPILE  LIB(SOURCE121) FIL(QRPGSRC) SRCMBR(TEST1) TYPE(SQLRPGLE) OPTIONS(*EVENTF)`

This is what I get from the above command.
![image](https://github.com/codefori/vscode-ibmi/assets/89184617/812b9009-b482-428d-b6ea-0c64e8dc0cba)

instead of something like this.
![image](https://github.com/codefori/vscode-ibmi/assets/89184617/89448128-4581-45cc-9ff8-62cfb6cd788f)

What changes do we need to do at our end so that the compile errors are captured and highlighted inline?

**Comment by worksofliam** [comment]
Hi @rishabkaushik 

Yes, we rely on the exit code of the call to determine a success or failure, which is similar to what all the compile commands (`CRTBNDRPG` for example) do also.

The spool is typically written to stdout also.

**Reply by rishabkaushik** [reply]
Thanks Liam, 

I am able to get the spool output in the Output tab but the errors are not highlighted probably because the action was successful in my case. I'll see what changes are needed at our end.

Thanks again!

**Reply by rishabkaushik** [reply]
Hey @worksofliam 

I did make some changes to the command so it returns an unsuccessful status if the compilation is unsuccessful. I can see the errors in Output tab as well(I was able to earlier as well), but when I click on the error it opens up a new editor for the same source I tried to compile and is already open. See below.

![Untitled](https://github.com/codefori/vscode-ibmi/assets/89184617/b1c9fd1c-29ca-4a2d-be20-21769abeccfb)

Any idea or suggestion on how do I fix this?

**Reply by sebjulliand** [reply]
Hi @rishabkaushik ,
can you hover with your cursor over both tab and tell us what is the path displayed in the tooltip?
![image](https://github.com/codefori/vscode-ibmi/assets/11096890/d13a7e9e-30f3-45d7-865e-64f6e655f1c1)

They may be different, in which case next step will be to figure out why there's a difference.

**Reply by chrjorgensen** [reply]
@sebjulliand Could it be because it is SQLRPGLE, where the SQL precompiler creates a temporary source?

**Reply by rishabkaushik** [reply]
Hi @sebjulliand ,

It is the same in both the members.

Thanks

**Comment by rishabkaushik** [comment]
> @sebjulliand Could it be because it is SQLRPGLE, where the SQL precompiler creates a temporary source?

Hey @chrjorgensen 

It is happening with RPGLE type as well, just checked.

**Comment by rishabkaushik** [comment, marked as answer]
This is no longer an issue. I can open the errors in the active window itself. I am on pre-release version if that helps anyone.
